src/PortingAssistant.Client.Porting/IPortingHandler.cs

```diff
@@ -15,12 +15,20 @@ public interface IPortingHandler
         /// <param name="projectPaths">List of projects paths</param>
         /// <param name="solutionPath">Path to solution file</param>
         /// <param name="targetFramework">Target framework to be used when porting</param>
+        /// <param name="originalVersions">List of key/value pairs where key is package and value is version number</param>
         /// <param name="upgradeVersions">List of key/value pairs where key is package and value is version number</param>
         /// <returns>A PortingResult object, representing the result of the porting operation</returns>
         List<PortingResult> ApplyPortProjectFileChanges(
             List<string> projectPaths,
             string solutionPath,
             string targetFramework,
             Dictionary<string, string> upgradeVersions);
+
+        List<PortingResult> ApplyPortProjectFileChanges(
+            List<string> projectPaths,
+            string solutionPath,
+            string targetFramework,
+            Dictionary<string, string> originalVersions,
+            Dictionary<string, string> upgradeVersions);
```

Comment:
How can we map a package to an upgraded version? We probably can use a tuple instead of two dictionaries:

List<Tuple<string, string, string, string>> versions = new List<Tuple<string, string, string, string>>();
                versions.Add(new Tuple<string, string, string, string>("OldPackage", "OldVersion", "NewPackage", "NewVersion"));

One other option would be creating an object in CTA that you can populate. Let me know if you want to discuss.